src/python/zquantum/core/measurement.py

```diff
@@ -500,7 +514,42 @@ def get_expectation_values(
                     value = -float(count) / num_measurements
                 expectation += np.real(coefficient) * value
             expectation_values.append(np.real(expectation))
-        return ExpectationValues(np.array(expectation_values))
+
+        correlations = np.zeros((len(ising_operator.terms),) * 2)
+        for i, first_term in enumerate(ising_operator.terms):
+            for j, second_term in enumerate(ising_operator.terms):
+                first_term_qubits = set(op[0] for op in first_term)
+                second_term_qubits = set(op[0] for op in second_term)
+                marked_qubits = first_term_qubits.symmetric_difference(
+                    second_term_qubits
+                )
+                for bitstring, count in bitstring_frequencies.items():
+                    bitstring_int = convert_bitstring_to_int(bitstring)
+                    if parity_even_p(bitstring_int, marked_qubits):
```

Comment:
Might be worth replacing `parity_even_p` from grove with our custom logic (see Slack)

Reply:
Also, I believe you are doing some unnecessary work when calculating correlations.  Diagonal elements of `correlations` are the squares of the `ising_operator` coefficients and the matrix itself is symmetric.

Reply:
I've modified the code to take advantage of symmetry and diagonal elements. @mstechly I'm going to put the replacement for `parity_even_p` in another PR because it gets used in places unrelated to this PR.

Reply:
Sure, sounds good for me!